ReactCommon/fabric/components/text/paragraph/ParagraphShadowNode.cpp

```diff
@@ -38,21 +38,18 @@ void ParagraphShadowNode::setMeasureCache(
   measureCache_ = cache;
 }
 
-void ParagraphShadowNode::updateLocalDataIfNeeded() {
+void ParagraphShadowNode::updateStateIfNeeded() {
   ensureUnsealed();
 
   auto attributedString = getAttributedString();
-  auto currentLocalData =
-      std::static_pointer_cast<const ParagraphLocalData>(getLocalData());
-  if (currentLocalData &&
-      currentLocalData->getAttributedString() == attributedString) {
+  auto state = getStateData();
+  if (state.getAttributedString() == attributedString) {
     return;
   }
 
-  auto localData = std::make_shared<ParagraphLocalData>();
-  localData->setAttributedString(std::move(attributedString));
-  localData->setTextLayoutManager(textLayoutManager_);
-  setLocalData(localData);
+  state.setAttributedString(std::move(attributedString));
```

Comment:
Following the previous comment, here we have to build a new state data object (only after we decided that we have to).